### DISCUSSION TITLE
Switching from one database to another

### DISCUSSION THREAD

**Discussion by ruebenacker** [prisma/prisma#7282] [Q&A] [answered]
Hey there, 

I'm currently exploring the possibilities of Prisma for a future project. It would be nice to know if it is possible to switch the chosen database later on and how easy the change/migration is. For example the first year or two use MySQL and after that switch to MS SQL or PostgreSQL. We are considering this for two reasons: 
1. Flexibility, not being locked in to one database 
2. Maybe later on we need the professional features and support offered by Microsoft for MS SQL Server

How much in the code needs to change? How easy is it to move the data from one database to another?

Would be nice to know. 
Thank you in advance!

**Comment by janpio** [comment, marked as answer]
Deploying the same Prisma Schema to different databases is quite easy. If you only use features that are available for all those databases, you most probably do not need to change any of your application code. 

If you use for example a scalar list, that is only available on PostgreSQL (https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference/#-modifier), you need to adapt your database and application of course.

The bigger issue will be to migrate your existing data from one database to another. Depending on the database size that might very well be done with a script that uses Prisma Client, but the bigger your database gets the more cumbersome (And risky) that will be, so you might want to use a specialized tool for that.
